app/controllers/PaypalController.scala

```diff
@@ -190,17 +192,22 @@ class PaypalController(ws: WSClient, paymentServices: PaymentServices, checkToke
     )(MetadataUpdate.apply)(MetadataUpdate.unapply)
   )
 
-  def updateMetadata(countryGroup: CountryGroup) = NoCacheAction.async(parse.form(metadataUpdateForm)) {
-    implicit request =>
-      val paypalService = paymentServices.paypalServiceFor(request)
-      val marketingOptIn = request.body.marketingOptIn
-      val idUser = IdentityId.fromRequest(request)
-      val contributor = request.session.data.get("email") match {
-        case Some(email) => paypalService.updateMarketingOptIn(email, marketingOptIn, idUser).value
-        case None => Future.successful(Logger.error("email not found in session while trying to update marketing opt in"))
-      }
-      contributor.map { _ =>
-        Redirect(routes.Contributions.thanks(countryGroup).url, SEE_OTHER)
-      }
+  def updateMetadata(countryGroup: CountryGroup) = NoCacheAction.async(parse.form(metadataUpdateForm)) { implicit request =>
+    val paypalService = paymentServices.paypalServiceFor(request)
+    val marketingOptIn = request.body.marketingOptIn
+    val idUser = IdentityId.fromRequest(request)
+    val contributor = request.session.data.get("email") match {
```

Comment:
I take good note of this one, but it's a bit out of scope as I really just touched the formatting here. I'll create another PR to surface your feedback